Add v3.0-preview APIs for LUIS

moabba opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-js

AutorestCI commented:
# Automation for azure-sdk-for-java

AutorestCI commented:
# Automation for azure-sdk-for-go

azuresdkci commented:
Can one of the admins verify this patch?

moabba committed:
Add v3.0-preview APIs

moabba committed:
Remove the timezoneoffset property

moabba committed:
Add the datetimeReference property

sarangan12 commented:
Since this is introducing a new version, we need to get approval from ARM team

sarangan12 commented:
I am fine with these changes. But, before merge ARM Team has to approve this PR (since this is a new version). So, I added appropriate label and wait for them.

moabba commented:
> Since this is introducing a new version, we need to get approval from ARM team

We already got an approval from the Azure review board, is that enough?

moabba commented:
Besides, the default PR description template says that this doesn't apply for date-plane APIs. Which I assume is our case because "data-plane" is in the path of our service swagger files.

sanjaiganesh commented:
ARM won't review data plane apis.. so removing the tag WaitForARMFeedback.

moabba committed:
- Add description to the definitions and the APIs
- Fix naming bugs (show-all-intents,dynamicLists)
- Rename the definitions for consistency
